app/controllers/application_controller.rb

```diff
@@ -450,6 +450,12 @@ def redirect_back_or_default(default, escape = true, use_escaped = true)
       begin
         uri = URI.parse(escaped_back_url)
 
+        # Remove basic auth credentials
+        uri.userinfo = nil
```

Comment:
It's a bit of a vague vector, as an attacker would still need to have some process to gather the returned status code in order to succesfully test credentials. 

I still think my reasoning is correct in that: We remove it and IF there is a use-case that we do not foresee, then we can still whitelist it again. I'm not clear why anyone would redirect to a basic auth, though.

Caching btw. differs from browser to browser, but I think all major vendors do that by default.